Include/internal/pycore_tagged.h

```diff
@@ -0,0 +1,73 @@
+#ifndef Py_INTERNAL_TAGGED_H
+#define Py_INTERNAL_TAGGED_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+#include <stddef.h>
+
+
+typedef union {
+    uintptr_t bits;
+} _Py_TaggedObject;
```

Comment:
Two comments on naming:

1) Most of these sorts of structs do not have an underscore after the `_Py` (e.g, `_PyInterpreterFrame`)
2) The `Object` suffix usually indicates a `PyObject` subtype, but this is something different (a tagged ptr)

Suggestions: `_PyTaggedPtr`, `_PyTaggedObjectPtr`, `_PyTaggedRef`